label-application-rule-config-service-impl/src/main/java/org/hypertrace/label/application/rule/config/service/LabelApplicationRuleValidatorImpl.java

```diff
@@ -264,6 +266,15 @@ public void validateLabelExpression(Action.DynamicLabel dynamicLabel) {
   private void validateTokenExtractionRule(
       Action.DynamicLabel.TokenExtractionRule tokenExtractionRule) {
     validateNonDefaultPresenceOrThrow(tokenExtractionRule, tokenExtractionRule.KEY_FIELD_NUMBER);
+    if (tokenExtractionRule.hasRegexCapture()) {
+      try {
+        Pattern.compile(tokenExtractionRule.getRegexCapture());
```

Comment:
I think we have a validator somewhere that validates a number of capture groups - and we should be using re2

Reply:
Oh we are :) I forget they use identical names to make it drop in. Anyway, `RegexValidator.validateCaptureGroupCount` is what I was referring to

Reply:
Done